pkg/raft/log.go

```diff
@@ -531,6 +550,27 @@ func (l *raftLog) scan(lo, hi uint64, pageSize entryEncodingSize, v func([]pb.En
 // The returned slice can be appended to, but the entries in it must not be
 // mutated.
 func (l *raftLog) slice(lo, hi uint64, maxSize entryEncodingSize) ([]pb.Entry, error) {
+	return l.snap().slice(lo, hi, maxSize)
+}
+
+// LogSlice returns a valid log slice for the (lo, hi] log index interval.
+func (l LogSnapshot) LogSlice(lo, hi uint64, maxSize uint64) logSlice {
```

Comment:
Is there a TODO here? `logSlice` is private, but I assume this is to be called from outside the package.

Reply:
Yeah, to be used outside the package. Possibly we won't need to access any of the `logSlice` outside `raft` package at first. The `MsgApp` sending code will pass in the returned `logSlice` back to raft without any modification, and `raft` will construct the message.

Added a TODO though.